src/App.tsx

```diff
@@ -0,0 +1,48 @@
+import { HashRouter, Route, Switch } from 'react-router-dom';
+import { ThemeProvider } from 'styled-components';
+
+import Header from './components/Header/Header';
+import GlobalStyle from './Global.styles';
+import ProductDetailPage from './pages/ProductDetailPage/ProductDetailPage';
+import ProductListPage from './pages/ProductListPage/ProductListPage';
+import ProductOrderPage from './pages/ProductOrderPage/ProductOrderPage';
+import ShoppingCartPage from './pages/ShoppingCartPage/ShoppingCartPage';
+import OrderListPage from './pages/OrderListPage/OrderListPage';
+import OrderDetailPage from './pages/OrderDetailPage/OrderDetailPage';
+
+import { PATH } from './constants';
+
+import { theme } from './App.styles';
+
+function App() {
+  return (
+    <ThemeProvider theme={theme}>
+      <GlobalStyle />
+      <HashRouter>
```

Comment:
> Bypassing the problem altogether: Hash History
With Hash History instead of Browser History, your URL for the about page would look something like this: http://example.com/#/about The part after the hash (#) symbol is not sent to the server. So the server only sees http://example.com/ and sends the index page as expected. React-Router will pick up the #/about part and show the correct page. [출처: React-router urls don't work when refreshing or writing manually](https://stackoverflow.com/questions/27928372/react-router-urls-dont-work-when-refreshing-or-writing-manually)

클라이언트 사이드 렌더링에서 Browser Router를 사용하면 리디렉션을 적용해도 새로고침을 하거나 주소창에 주소를 입력하는 경우 404 Error가 뜨게 됩니다. 반면 HashRouter를 쓰게 되면 # 이후의 path는 무시되기 때문에 React Router가 정상적으로 동작합니다. 물론, SEO를 생각한다면 BrowserRouter가 권장되겠지만 현재로서는 서버 사이드 렌더링과 클라이언트 사이드 렌더링을 섞을 만한 능력이 없어 HashRouter를 사용했습니다 😂